nibabel/cifti2/parse_cifti2_fast.py

```diff
@@ -0,0 +1,585 @@
+        return self._content
+
+    def _mangle(self, value):
+        if not isinstance(value, Cifti2Header):
+            raise ValueError('Can only mangle a Cifti2Header.')
+        return value.to_xml()
+
+extension_codes.add_codes((
+    (Cifti2Extension.code, 'cifti', Cifti2Extension),))
+
+intent_codes.add_codes((
+    # The codes below appear on the CIFTI-2 standard
+    # http://www.nitrc.org/plugins/mwiki/index.php/cifti:ConnectivityMatrixFileFormats
+    # https://www.nitrc.org/forum/attachment.php?attachid=341&group_id=454&forum_id=1955
+    (3000, 'dense fiber/fan samples', (), 'NIFTI_INTENT_CONNECTIVITY_UNKNOWN'),
+    (3001, 'dense connectivity', (), 'NIFTI_INTENT_CONNECTIVITY_DENSE'),
+    (3002, 'dense data series/fiber fans', (),
+     'NIFTI_INTENT_CONNECTIVITY_DENSE_SERIES'),
+    (3003, 'parcellated connectivity', (),
+     'NIFTI_INTENT_CONNECTIVITY_PARCELLATED'),
+    (3004, 'parcellated data series', (),
+     "NIFTI_INTENT_CONNECTIVITY_PARCELLATED_SERIES"),
+    (3006, 'dense scalar', (),
+     'NIFTI_INTENT_CONNECTIVITY_DENSE_SCALARS'),
+    (3007, 'dense label', (),
+     'NIFTI_INTENT_CONNECTIVITY_DENSE_LABELS'),
+    (3008, 'parcellated scalar', (),
+     'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_SCALAR'),
+    (3009, 'parcellated dense connectivity', (),
+     'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_DENSE'),
+    (3010, 'dense parcellated connectivity', (),
+     'NIFTI_INTENT_CONNECTIVITY_DENSE_PARCELLATED'),
+    (3011, 'parcellated connectivity series', (),
+     'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_PARCELLATED_SERIES'),
+    (3012, 'parcellated connectivity scalar', (),
+     'NIFTI_INTENT_CONNECTIVITY_PARCELLATED_PARCELLATED_SCALAR')))
+
+
+class _Cifti2AsNiftiHeader(Nifti2Header):
+    ''' Class for Cifti2 header extension '''
+
+    @classmethod
+    def _valid_intent_code(klass, intent_code):
+        """ Return True if `intent_code` matches our class `klass`
+        """
+        return intent_code >= 3000 and intent_code < 3100
+
+    @classmethod
+    def may_contain_header(klass, binaryblock):
+        if not super(_Cifti2AsNiftiHeader, klass).may_contain_header(binaryblock):
+            return False
+        hdr = klass(binaryblock=binaryblock[:klass.sizeof_hdr])
+        return klass._valid_intent_code(hdr.get_intent('code')[0])
+
+    @classmethod
+    def _get_checks(klass):
+        # We need to return our own versions of - e.g. chk_datatype, to
+        # pick up the Nifti datatypes from our class
+        return (klass._chk_sizeof_hdr,
+                klass._chk_datatype,
+                klass._chk_bitpix,
+                klass._chk_pixdims,
+                klass._chk_magic,
+                klass._chk_offset,
+                klass._chk_qform_code,
+                klass._chk_sform_code)
+
+    def _chk_pixdims(hdr, fix=False):
+        rep = Report(CIFTI2HeaderError)
+        pixdims = hdr['pixdim']
+        spat_dims = pixdims[1:4]
+        if not np.any(spat_dims <= 0):
```

Comment:
Oops - so we did - my mistake - that should be `spat_dims < 0`.   I'll add a PR and test.